hbase-shaded-protobuf/src/main/patches/HBASE-15789_V2.patch

```diff
@@ -1,5 +1,5 @@
 diff --git a/src/main/java/com/google/protobuf/ByteBufferWriter.java b/src/main/java/com/google/protobuf/ByteBufferWriter.java
-index 2cb3ada67..6e1737c45 100644
+index 3970b0ea5..571b0a975 100644
 --- a/src/main/java/com/google/protobuf/ByteBufferWriter.java
```

Comment:
Not a review comment but a query, I am not aware of why these patch files are maintained? Could you please share why/how this change is made for my understanding?

Reply:
We apply some patches when shading protobuf-java, so when upgrading the version for protobuf, we also need to update the patch file otherwise the maven build may fail because can not apply patches.

I do not know how others deal with these patches, for me, I will checkout the protobuf code base and apply these patches to the code base, resolve the conflicts, and re-generate the patches again.

Reply:
Thanks a lot for the detailed explanation. :)